### PR TITLE
Orog gsl improvements

### PR DESCRIPTION
**Pull request by mdtoyNOAA**
[OA3_fixes.pdf](https://github.com/ufs-community/UFS_UTILS/files/10147776/OA3_fixes.pdf)
## DESCRIPTION OF CHANGES: 
The orographic asymmetry (OA) fields generated by orog_gsl have a negative bias, especially for high resolution grids. The issue is that with higher-resolution grids, there are fewer fine-grid (5km spacing for large-scale fields, and 1km for small-scale fields) topographic data points within each model grid cell to use for sampling and determining the statistical properties, such as OA. Songyou Hong (NOAA/PSL) discovered that the cause of the negative bias is due to the case when there are an odd number of fine-grid points along a cardinal direction relative to the model grid cell. The limits of the "do-loops" for OA calculation cause fewer points to be sampled in one-half of the grid cell than the other. Songyou found a fix to this problem, which is to modify the limits of the "do-loops" when there are an odd number of fine-grid points along a cardinal direction so that the number of points in each half of the grid are the same.  Details of the fix are described in issue [#716](https://github.com/ufs-community/UFS_UTILS/issues/716).
Plots of the orographic asymmetry in the South-West direction (OA3) are shown [here](https://github.com/ufs-community/UFS_UTILS/files/10147776/OA3_fixes.pdf).  The old baseline fields show a negative bias (too much blue shading), especially for the "large-scale" field, which uses a coarser sub-grid topographic data set and results in larger sampling error.  The new fields display a more random field, which varies between -1 and 1, which is the desired result.

## TESTS CONDUCTED: 
Contingency tests were run on Hera with the Intel compiler.  The tests were run with the script UFS_UTILS/reg_tests/grid_gen/driver.hera.sh.  All tests passed except for the "REGIONAL GSL GWD TEST", which is expected because this PR changes the baseline results.  The test log "consistency.log05" is on Hera at:
/scratch1/BMC/wrfruc/mtoy/git_local/UFS_UTILS/reg_tests/grid_gen

The differences from the baseline were only due to the "OA1,OA2,OA3,OA4" variables, as expected.  The following lines from "consistency.log05 show this:
+ /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C772_oro_data_ls.tile7.halo0.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/regional.gsl.gwd/C772_oro_data_ls.tile7.halo0.nc
Variable Group Count     Sum  AbsSum       Min Max    Range     Mean   StdDev
oa1      /     21981 8205.24 8205.24 0.0357143   1 0.964286 0.373288 0.165789
oa2      /     37232 13395.9 13395.9  0.047619   1 0.952381 0.359795 0.153006
oa3      /     49106 42914.1 42914.1  0.178571   2  1.82143 0.873908 0.268177
oa4      /     47601 18064.9 18064.9 0.0357143   1 0.964286 0.379508 0.179291

+ /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/intel-2022.1.2/nccmp/1.8.9.0/bin/nccmp -dmfqS C772_oro_data_ss.tile7.halo0.nc /scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/regional.gsl.gwd/C772_oro_data_ss.tile7.halo0.nc
Variable Group Count     Sum  AbsSum         Min      Max    Range      Mean    StdDev
oa1      /     21914 1305.27 1305.27 0.000137746 0.236949 0.236811 0.0595634 0.0332465
oa2      /       794 62.2958 62.2958 0.000238895 0.198674 0.198435 0.0784582 0.0375766
oa3      /     48203 10020.6 10020.6 0.000925064 0.675325   0.6744  0.207883 0.0718323
oa4      /     48157    3474    3474 0.000163794 0.302521 0.302357  0.072139 0.0354685

The new baselines are on Hera at:
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/regional.gsl.gwd/C772_oro_data_ls.tile7.halo0.nc
and
/scratch1/NCEPDEV/nems/role.ufsutils/ufs_utils/reg_tests/grid_gen/baseline_data/regional.gsl.gwd/C772_oro_data_ss.tile7.halo0.nc

## DEPENDENCIES:
None

## DOCUMENTATION:
N/A

## ISSUE: 
This PR fixes issue [#716](https://github.com/ufs-community/UFS_UTILS/issues/716).

## CONTRIBUTORS: 
Thank you to Songyou Hong (songyou.hong@noaa.gov) (NOAA/PSL) for contributing the code fix.

### CODE REVIEW

File: sorc/orog_mask_tools.fd/orog_gsl.fd/module_gsl_oro_data_lg_scale.f90 [hunk 1]
```diff
@@ -611,9 +611,15 @@ subroutine calc_gsl_oro_data_lg_scale(tile_num,res_indx,halo)
...
       do jj = 1,jj_m
-         do ii = 1,ii_m/2   ! left half of box
-            if ( zs(ii,jj) > zs_mean ) nu = nu + 1
-         end do
+         if(mod(ii_m,2).eq.0.) then
+           do ii = 1,ii_m/2   ! left half of box
+              if ( zs(ii,jj) > zs_mean ) nu = nu + 1
+           end do
+         else
+           do ii = 1,ii_m/2+1   ! left half of box
```
Comment (GeorgeGayno-NOAA) [inline sorc/orog_mask_tools.fd/orog_gsl.fd/module_gsl_oro_data_lg_scale.f90:619]: Just curious about your solution. It looks like you are including the column of points at (ii_m/2+1) in the both the left and right half of the box?

Comment (mdtoyNOAA) [reply sorc/orog_mask_tools.fd/orog_gsl.fd/module_gsl_oro_data_lg_scale.f90:619]: Yes, that is correct.  And this occurs only when thera are an odd number of fine grid points within a coarse model grid cell.  The alternative would be not to count the center column at all in either left or right halves of the box.  The key is to have the same number of points counted in each half.

**Comment by GeorgeGayno-NOAA** [comment]
@mdtoyNOAA - In the PR description, you say, _"Plots of the orographic asymmetry in the South-West direction (OA3) are shown on the attached plot."_ Can you link to the plot or the pdf containing the plot? Or say something like, _"See the Mike Toy presentation in issue #716."_

**Comment by mdtoyNOAA** [comment]
@GeorgeGayno-NOAA   Will do.  I attached it using the generic "add file" button at the bottom of the edit page.  It put the link at the top.  I was afraid folks would miss it.

**Comment by GeorgeGayno-NOAA** [comment]
Checked out the branch at 17d39be on Jet and Orion. The branch was compiled and the `grid_gen` regression tests were run. The REGIONAL GSL GWD TEST failed as expected. The differences from the baseline were the same as on Hera. All other `grid_gen` tests passed. Will test on WCOSS2 when that machine is out of maintenance.

**Comment by GeorgeGayno-NOAA** [comment]
> Checked out the branch at [17d39be](https://github.com/ufs-community/UFS_UTILS/commit/17d39be93407e8246dd5c3796ddf02a02b1e6b31) on Jet and Orion. The branch was compiled and the `grid_gen` regression tests were run. The REGIONAL GSL GWD TEST failed as expected. The differences from the baseline were the same as on Hera. All other `grid_gen` tests passed. Will test on WCOSS2 when that machine is out of maintenance.

The above was repeated on Hera and WCOSS2. The same test failed. The difference from the baseline on both machines matched that documented by @mdtoyNOAA.

Will merge to 'develop'.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**